torch/onnx/_internal/fx/diagnostics.py

```diff
@@ -71,12 +78,52 @@ def _torch_fx_graph_module(obj: torch.fx.GraphModule) -> str:
 
 @_format_argument.register
 def _torch_fx_node(obj: torch.fx.Node) -> str:
-    return f"fx.Node({obj.name}[{obj.op}]'{obj.target}')"
+    node_string = f"fx.Node({obj.target})[{obj.op}]:"
+    if "val" not in obj.meta:
+        return node_string + "None"
+    return node_string + _format_nested_argument_by_dtype(obj.meta["val"])
+
+
+@_format_argument.register
+def _torch_fx_symbolic_value(
+    obj,  # NOTE: functools.singledispatch does not support Union until 3.11, so we use Any here.
+) -> str:
+    return f"Sym({obj})"
 
 
 @_format_argument.register
 def _torch_tensor(obj: torch.Tensor) -> str:
-    return f"Tensor(shape={obj.shape}, dtype={obj.dtype})"
+    return f"Tensor({fx_type_utils.from_torch_dtype_to_abbr(obj.dtype)}{_stringify_shape(obj.shape)})"
+
+
+@_format_argument.register
+def _list(obj: list) -> str:
+    list_string = f"List[length={len(obj)}](\n"
+    if not obj:
+        return list_string + "None)"
+    for item in obj:
+        list_string += f"{_format_nested_argument_by_dtype(item)},\n"
+    return list_string + ")"
+
+
+@_format_argument.register
+def _tuple(obj: tuple) -> str:
+    tuple_string = f"Tuple[length={len(obj)}](\n"
+    if not obj:
+        return tuple_string + "None)"
+    for item in obj:
+        tuple_string += f"{_format_nested_argument_by_dtype(item)},\n"
+    return tuple_string + ")"
+
+
+@_format_argument.register
+def _dict(obj: dict) -> str:
+    dict_string = f"Dict[length={len(obj)}](\n"
+    if not obj:
+        return dict_string + "None)"
+    for key, value in obj.items():
+        dict_string += f"{key}: {_format_nested_argument_by_dtype(value)},\n"
```

Comment:
```suggestion
        dict_string += f"{key}: {format_argument(value)},\n"
```

I think this should be equivalent ?

Reply:
But if it's nested inside value, `format_argument` won't be able to unpack it.